src/hotspot/cpu/x86/x86.ad

```diff
@@ -9451,64 +9451,32 @@ instruct evcmp_masked(kReg dst, vec src1, vec src2, immI8 cond, kReg mask, rRegP
   ins_pipe( pipe_slow );
 %}
 
-#ifdef _LP64
-instruct mask_all_evexI_imm(kReg dst, immI cnt, rRegL tmp) %{
+instruct mask_all_evexI_imm_LE32(kReg dst, immI cnt, rRegI tmp) %{
+  predicate(Matcher::vector_length(n) <= 32);
   match(Set dst (MaskAll cnt));
-  effect(TEMP_DEF dst, TEMP tmp);
-  format %{ "mask_all_evexI $dst, $cnt \t! using $tmp as TEMP" %}
-  ins_encode %{
-    int vec_len = Matcher::vector_length(this);
-    if (VM_Version::supports_avx512bw()) {
-      __ movq($tmp$$Register, $cnt$$constant);
-      __ kmovql($dst$$KRegister, $tmp$$Register);
-      __ kshiftrql($dst$$KRegister, $dst$$KRegister, 64 - vec_len);
-    } else {
-      assert(vec_len <= 16, "");
-      __ movq($tmp$$Register, $cnt$$constant);
-      __ kmovwl($dst$$KRegister, $tmp$$Register);
-      __ kshiftrwl($dst$$KRegister, $dst$$KRegister, 16 - vec_len);
-    }
-  %}
-  ins_pipe( pipe_slow );
-%}
-
-instruct mask_all_evexI(kReg dst, rRegI src, rRegL tmp) %{
-  match(Set dst (MaskAll src));
-  effect(TEMP_DEF dst, TEMP tmp);
-  format %{ "mask_all_evexI $dst, $src \t! using $tmp as TEMP" %}
+  effect(TEMP dst, TEMP tmp);
```

Comment:
TEMP dst is not needed here.

Reply:
dst is both read and written to in macro assembly routine.

Reply:
dst both written and read is not the criteria for TEMP dst effect. Only if the src is used after dest is written in the encoding for instruct, TEMP dst is needed.

Reply:
Hi, do we need `TEMP dst` if some `tmp` is used after `dst` is written, as without it `tmp` and `dst` are implied to have non-overlapping lifetime and can lead to conflict. This is a pure question unrelated to the PR. Thank you very much.

Reply:
Hi @merykitty, yes we would need TEMP dst in that case as well.

Reply:
TEMP attribute ensures creation of a temporary machine operand which interferes with source operands if their registerclasses are overlapping, a TEMP + DST data flow attribute ensures that DST is not accidentally allocated a SRC register even SRC is not live beyond that instruction. This shall prevent overriding the src before using its value the instruction encoding block.